### PR TITLE
MINOR: remove unused MeteredKeyValueStore

### PR DESCRIPTION
**Pull request by mjsax**
- remove MeteredKeyValueStore
- rename MeteredKeyValueByteStore to MeteredKeyValueStore
- code cleanup (Java8 and others)
- removed usage of deprecated `KafkaMetric#value()`
- merged `InnerMeteredKeyValueStore` into `MeteredKeyValueStore` and removed `TypeConverter` interface

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck

**Comment by mjsax** [comment]
Working on KIP-258 is discovered this. The diff is a little weird, because I also renamed `MeteredKeyValueByteStore` to `MeteredKeyValueStore` (and removed old `MeteredKeyValueStore`)

**Review by guozhangwang — COMMENTED** [review]
It seems that you've actually renamed `MeteredKeyValueBytesStore` to `MeteredKeyValueStore`, and removed the original `MeteredKeyValueStore` right? In this case should we  rename `MeteredKeyValueBytesStoreTest` to `MeteredKeyValueStoreTest`?

EDIT: actually nvm, realized you've already done this.

**Comment by guozhangwang** [comment]
LGTM.

**Comment by mjsax** [comment]
Update this PR to merged `InnerMeteredKeyValueStore` into `MeteredKeyValueStore` and to removed `TypeConverter` interface

**Review by bbejeck — APPROVED** [review]
Thanks for the patch LGTM.

File: streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java [hunk 1]
```diff
@@ -16,114 +16,328 @@
...
+
+    private V outerValue(final byte[] value) {
+        return value == null ? null : serdes.valueFrom(value);
+    }
+
+    private List<KeyValue<Bytes, byte[]>> innerEntries(final List<KeyValue<K, V>> from) {
+        final List<KeyValue<Bytes, byte[]>> byteEntries = new ArrayList<>();
+        for (final KeyValue<K, V> entry : from) {
+            byteEntries.add(KeyValue.pair(Bytes.wrap(serdes.rawKey(entry.key)), serdes.rawValue(entry.value)));
+
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java]: nit: empty line.

**Review by guozhangwang — APPROVED** [review]
One nit comment, otherwise lgtm.

Please feel free to merge after jenkins passed.

**Comment by mjsax** [comment]
Merged to `trunk`.

**Pull request merged**